packages/maskbook/src/web3/hooks/useContract.ts

```diff
@@ -33,6 +34,7 @@ function createContract<T extends Contract>(from: string, address: string, ABI:
                                 from,
                                 to: contract.options.address,
                                 data: cached.encodeABI(),
+                                gasPrice: currentGasPriceSettings.value,
```

Comment:
The initial amount of `currentGasPriceSettings` is zero. You should make sure that you have already set it properly. Maybe a run time reader function is better. Then you can handle complex logic in that.